code/modules/paperwork/prefab_papers.dm

```diff
@@ -0,0 +1,235 @@
+/datum/prefab_document
+	var/contents
+
+GLOBAL_REFERENCE_LIST_INDEXED(prefab_papers, /obj/item/paper/prefab, document_title)
+/obj/structure/filingcabinet/documentation
+	name = "documents cabinet"
+	color = "#e78738"
+
+	/// The categories available from this cabinet. Manually populated.
+	var/list/available_categories = list()
+	/// The possible options that can be withdrawn. Automatically populated by get_possible_documents().
+	var/list/available_documents = list()
+	/// Number of prefab documents that can be withdrawn. Cap intended to prevent people spamming infinite copies.
+	var/remaining_documents = 30
+
+/obj/structure/filingcabinet/documentation/Initialize()
+	. = ..()
+	get_possible_documents()
+
+/obj/structure/filingcabinet/documentation/proc/get_possible_documents()
+	available_documents.Cut()
+	for(var/docname in GLOB.prefab_papers)
+		var/obj/item/paper/prefab/document = GLOB.prefab_papers[docname]
+		if(!istype(document))
+			continue
+		log_debug("DOCUMENT: Checking [document]")
+		if(!document.is_prefab || !document.doc_datum_type || (document.name == "paper"))
+			log_debug("DOCUMENT: Prefab, Name or Ref fail.")
+			continue
+		if(!document.document_category || !(document.document_category in available_categories))
+			log_debug("DOCUMENT: Category fail.")
+			continue
+		available_documents += docname
+	return
```

Comment:
This has many that report failures. See any game log where this is TM'd. It also doesn't need to be repeated for every filingcabinet on the map. Rather have this proc ran once to initialize say a GLOB so the checks are ran once. Nothing here appears to be specific to the filing cabinet its being run for.

Reply:
I should have removed these a long time ago, I haven't needed them since near the start of the PR when I couldn't figure out why things weren't working.